src/commands.rs

```diff
@@ -1,11 +1,29 @@
-use crate::state_machine::{CharacterSet, StateMachine};
+use crate::{
+    api,
+    state_machine::{CharacterSet, StateMachine},
+};
 use reqwest::blocking::Client as HttpClient;
 use serenity::{model::channel::Message, prelude::Context};
 use std::{collections::HashMap, sync::Arc};
 
 const PREFIX: &'static str = "?";
 pub(crate) type Result<T> = std::result::Result<T, Box<dyn std::error::Error>>;
-pub(crate) type CmdPtr = Arc<dyn for<'m> Fn(Args<'m>) -> Result<()> + Send + Sync>;
+pub(crate) type GuardFn = fn(&Args) -> Result<bool>;
+
+struct Command {
+    guard: GuardFn,
+    ptr: Box<dyn for<'m> Fn(Args<'m>) -> Result<()> + Send + Sync>,
```

Comment:
Should this also get a type alias?

Reply:
This is the only place where that signature is used.  :woman_shrugging: its not really worth it IMO

Reply:
oh yeah i forgot we can't do that yet...